src/component/MainView.tsx

```diff
@@ -64,6 +65,7 @@ export class MainView extends React.Component<MainViewProps, MainViewState> {
 
   public componentDidMount(): void {
     if (this.props.user === EMPTY_USER) {
+      Routing.saveOriginalTarget();
```

Comment:
Could just confirm that this is because Keycloak-based backend returns 302 forward to Keycloak login instead of 401? Because for 401 responses this is handled by Ajax.